projects/.metadata/.plugins/org.eclipse.core.resources/.history/25/00a1e777f21d00181353d905c66b77c4

```diff
@@ -0,0 +1,384 @@
+
+/*!
+ * @brief      Initializes the ESP8266 Wi-Fi module.
+ *
+ * @details    Initialize the ESP8266 Wi-Fi module and test the
+ *             hardware connectivity. Try to connect to an access
+ *             point and establish a TCP connection.
+ */
+void InitWiFiConnection()
+{
+	ADI_WIFI_RESULT eResult;
+
+    /* Initialize the ESP8266 Wi-Fi module */
+	eResult = adi_wifi_Init(adi_wifi_AplicationCallback, NULL);
+    DEBUG_RESULT("Error initializing the ESP8266 module.\r\n", eResult, ADI_WIFI_SUCCESS);
+
+	eResult = adi_wifi_radio_TestAT();
+    DEBUG_RESULT("Error testing hardware connectivity.\r\n", eResult, ADI_WIFI_SUCCESS);
+
+	eResult = adi_wifi_radio_SetWiFiMode(1);
+    DEBUG_RESULT("Error establishing a tcp connection.\r\n", eResult, ADI_WIFI_SUCCESS);
+
+	eResult = adi_wifi_radio_DisconnectFromAP();
+    DEBUG_RESULT("Error disconnecting from access point.\r\n", eResult, ADI_WIFI_SUCCESS);
+
+	eResult = adi_wifi_radio_ConnectToAP(aWifiSSID, aWifiPassword,  NULL);
+    DEBUG_RESULT("Error connecting to access point.\r\n", eResult, ADI_WIFI_SUCCESS);
+
+	eResult = adi_wifi_radio_EstablishTCPConnection(&gTCPConnect);
+    DEBUG_RESULT("Error establishing a tcp connection.\r\n", eResult, ADI_WIFI_SUCCESS);
+}
+
+/*!
+ * @brief      Initializes the system
+ *
+ * @details    This function is responsible for initializing the pinmuxing, power service
+ *             and wifi subsystem. It also initializes the realtime clock for to timestamp
+ *             the outgoing sensor data packets.
+ */
+static void InitSystem(void)
+{
+    ADI_PWR_RESULT  ePwr;
+
+    /* Explicitly disable the watchdog timer */
+    *pREG_WDT0_CTL = 0x0u;
+
+    /* Pinmux */
+    adi_initpinmux();
+
+    /* Initialize clocks */
+    ePwr = adi_pwr_Init();
+    DEBUG_RESULT("Error initializing the power service.\r\n", ePwr, ADI_PWR_SUCCESS);
+
+    ePwr = adi_pwr_SetClockDivider(ADI_CLOCK_HCLK, 1u);
+    DEBUG_RESULT("Error configuring the core clock.\r\n", ePwr, ADI_PWR_SUCCESS);
+
+    ePwr = adi_pwr_SetClockDivider(ADI_CLOCK_PCLK, 1u);
+    DEBUG_RESULT("Error configuring the peripheral clock.\r\n", ePwr, ADI_PWR_SUCCESS);
+
+    /* Init timestamping */
+    INIT_TIME();
+
+    timer_start();
+
+    DEBUG_MESSAGE("Starting MQTT publisher Wi-Fi ADXL362 accelerometer demo application\r\n");
+
+}
+
+/*!
+ * @brief      Troubleshoot and attempt to fix connection
+ *
+ * @details    This function is called if the ESP8266 has failed to communicate with the AP or broker.
+ *             It will attempt to diagnose and repair the connectivity problem.
+ */
+static void RepairConnection()
+{
+    uint8_t         nConnectionStatus;
+
+	if(adi_wifi_radio_GetConnectionStatus(&nConnectionStatus) == ADI_WIFI_SUCCESS)
+		switch(nConnectionStatus) {
+			/*The  Station is connected to an AP and its IP is obtained, so try to connect to broker. */
+			case 2:
+				/* Send MQTT "CONNECT" message to broker */
+				if(adi_wifi_radio_MQTTConnect(&gMQTTConnect) != ADI_WIFI_SUCCESS)
```

Comment:
case should be under the switch, you have 1 extra tab, it should be something like
```
switch (action) {
case KOBJ_ADD:
        return "add";
case KOBJ_REMOVE:
        return "remove";
case KOBJ_CHANGE:
        return "change";
default:
        return NULL;
}
```

take a look at: https://www.kernel.org/doc/html/v4.10/process/coding-style.html#placing-braces-and-spaces